exist-core/src/main/java/org/exist/storage/BrokerPool.java

```diff
@@ -975,29 +974,25 @@ public NotificationService getNotificationService() {
      * @return <code>true</code> if transactions can be handled
      */
     public boolean isRecoveryEnabled() {
-        synchronized(readOnly) {
-            return !readOnly && recoveryEnabled;
-        }
+        return !readOnly.get() && recoveryEnabled;
     }
 
     @Override
     public boolean isReadOnly() {
-        synchronized(readOnly) {
-            if(!readOnly) {
-                final long freeSpace = FileUtils.measureFileStore(dataLock.getFile(), FileStore::getUsableSpace);
-                if (freeSpace != -1 && freeSpace < diskSpaceMin) {
-                    LOG.fatal("Partition containing DATA_DIR: {} is running out of disk space [minimum: {} free: {}]. Switching eXist-db into read-only mode to prevent data loss!", dataLock.getFile().toAbsolutePath().toString(), diskSpaceMin, freeSpace);
-                    setReadOnly();
-                }
+        final boolean isReadOnly = readOnly.get();
+        if (!isReadOnly) {
```

Comment:
I think this local variable is only used here and can be inlined

Reply:
I felt it was clearer to read in the way I formulated it. How strong is your objection?

Reply:
for debugging is is more clear; I'd not care

Reply:
@adamretter just a hint, not a requirement..